testing/functional_tests/test_automate_training.py

```diff
@@ -34,6 +35,38 @@ def test_automate_training(download_functional_test_files, script_runner):
     check_sha256(str(file_config))
 
 
+def test_automate_training_run_test_debug(download_functional_test_files):
+    """A unit test similar to test_automate_training_run_test but allow step through (instead of using script caller/
+    subprocess mode which cannot be stepped. Other than that, nothing else really changed and is exactly the same.
+    Very useful for debugging this high level function to spot problems
+
+    Fixture Required:
+        download_functional_test_files:
+    """
```

Comment:
New unit test which help debug/step the automated_training end to end run process.

Reply:
Being split into clean up PR.